### ISSUE TITLE
Instrument PCs: Log rotation automation

### ISSUE THREAD

**Issue by GRyall** [ISISComputingGroup/IBEX#8360]
As a systems administrator I would like there to be an automated process that moves older logs off instrument PCs to free up space. Currently this consumes support time either responding to alerts for full disks or manually running many processes between cycle. 

Additionally, I would like conserver log files on Instrument PCs older than 10 days to be deleted, this will reduce effort spent freeing space on instrument PCs and match the period other logs are kept before being moved off instrument PCs.
### Acceptance Criteria

- [x] #8361 has been completed
- [x] An automated process exists which moves log files older than 10 days (except the conserver log and MySQL logs) off the instrument PCs in a similar fashion as @ChrisM-S  current manually run script. 
- [x] An automated process is in place that deletes conserver logs older than 10 days
- [x] Appropriate checks are in place that this script is running appropriately
- [x] The process is documented in the developer wiki

### Extra Information

We have other tasks already running at regular interviews, see e.g. our git processes in the blockserver.

This ticket is a result of the discussion that took place as part of #8321 .

See also: [developer wiki page](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Data-Generation-and-Storage-on-Instrument-PCs-(NDX's)).

Separate ticket exists for the mysql related logs (#8364).

It is recognised that the Nagios alerts for full disks are currently used as a proxy fie the health of the controls system on the instrument PC (as an unhealthy system generates large logs). To that end this ticket depends on #8361  (Ticket on Nagios check monitoring log growth/size).

## How to Test
_verbose instructions for reviewer to test changes
(**Add before making a PR**)_

time in planning 01:30 23/5/24
~00:29 20/06/24

**Comment by isaachilly** [comment]
Note that the automatic removal of logs will require modification of the IOC Log File Rate service on Nagios made in:
- https://github.com/ISISComputingGroup/IBEX/issues/8361

@FreddieAkeroyd and me (@isaachilly ) talked about what would be best. Problem: If interval the service uses as a start and end time to get difference in log dir size encompasses a time when logs are taken of this will create an inaccurate log rate. Solution (for now, and at first glance) is that the service should raise an alert if the logging rate is theoretically larger than we can take off for the critical level and for warning would be just an abnormal logging rate for an inst.

**Comment by Tom-Willemsen** [comment]
PRs:
- https://github.com/ISISComputingGroup/IBEX/pull/8449
- https://github.com/ISISComputingGroup/EPICS/pull/360
- https://github.com/ISISComputingGroup/ibex_utils/pull/200

## To review

Note: you cannot review this ticket on a developer machine, as your developer account does not have write access to the `stage-deleted` area. You will need to review this ticket on an instrument-like machine (e.g. SCIDEMO or a similar machine).

SCIDEMO does have a few logfiles that can't be moved as they're owned by admin, these can conveniently be used to check the "dry-run" nagios check below. As developers if this happened on a real instrument nagios would then notify us and we can intervene remotely to remove these admin-owned files.

You can also go into windows task scheduler and manually trigger the task if you don't fancy waiting overnight for it to run

## Once ticket is merged

I don't know if this can be deployed easily remotely - it needs admin to add scheduled task, we may need to deploy it "manually" to all instruments next shutdown - see task in tasks board

## Once ticket is deployed on all relevant instruments

Ask Freddie to add `ssh <user>@<inst> c:\instrument\apps\python3\python.exe c:\instrument\apps\epics\utils\logrotate.py --time 14 --dry-run` as a nagios check. 

Files should be being deleted/moved after 10 days, the nagios check will then fail if they still haven't been moved at 14 days old (this gives a few opportunities to retry the next day in case e.g. network goes down)

See tasks in task board

**Comment by Chsudeepta** [comment]
Verified on SCIDEMO following the steps mentioned. The job is created on the task scheduler as expected. We now have to wait for 10 days to verify that it is working on the expected lines.

**Comment by Chsudeepta** [comment]
Copied files more than 10 days old to "var" directories. Found them deleted from conserver folder and those in the deploy folder moved to \\isis\inst$\Backups$\stage-deleted\ndxSCIDEMO\deploy. So, it seems to be working as expected.
